libs/nss.c

```diff
@@ -124,6 +145,10 @@ int _nss_tcb_getspent_r(struct spwd *__result_buf,
 	} while (!strcmp(result->d_name, ".") ||
 	    !strcmp(result->d_name, "..") || result->d_name[0] == ':');
 
+#if USE_READDIR
+	errno = saved_errno;
+#endif
+
```

Comment:
Please move it inside the loop where `saved_errno = errno;` is made, otherwise `saved_errno` may be assigned to the value of `errno` clobbered in the previous loop iteration.

Reply:
Fixed in rebased commit.